docs/dev/plugins/cloudFoundry/Transformation_Process.md

```diff
@@ -0,0 +1,151 @@
+# Cloud Foundry - Transformation Process
+
+The purpose of this document is to show the transformation logic and process of the CloudFoundry plugin for a developer.
+This document does not mention all classes which are used in the java code but describes the method of the transformation so that the developer is able to understand the code and the transformation process.   
+This is the text based documentation, you can find an example transformation with some pictures [here](Transformation_example.pdf).
+
+## Foreword
+- the TOSCA model is a graph. So we are talking about some graph terms. For further information please read the documentation of the core/model.
+- plugin gets some properties (at the beginning of transformation) from the user to create connection to CloudFoundry instance.
+- plugin is using visitor patterns for nodes. The visitors contain the logic for the node types.
+- there will be created one manifest for all applications in the model
+- per application there is one CloudFoundry application object (cf-app) which contains all information of the application
+- the output is called target artifact
+
+## Visitors
+### Node visitor
+The node visitor contains the logic for the supported node types. It fills the cf-app with all information out of the node.
```

Comment:
Consider using the actual class names `NodeVisitor` and `PrepareVisitor` at some points for better understanding.